### PR TITLE
mark demo responses as demos with train task idxs

### PR DESCRIPTION
**Pull request by tomsilver**
demonstration responses should be marked as demos. we also will need to know at least the goal that the trajectories are for. one thing to be aware of in general is: now, if a LowLevelTrajectory has train_task_idx 5, it does _not_ mean that the trajectory is a full demonstration of the entire train_task_idx 5; it just means that the trajectory ends up in the goal for that train task, starting at a potentially arbitrary state. if this is too confusing, we may need to make `is_demo`, `train_task_idx`, and `goal` three different properties for `LowLevelTrajectory`. @ronuchit lmk what you think -- I am neutral about whether we should do that.

### CODE REVIEW

File: tests/test_teacher.py
Comment (tomsilver) [inline tests/test_teacher.py:71]: This is the assertion that matters -- the other changes in this PR are just to make this assertion pass

**Review by NishanthJKumar — COMMENTED** [review]
LGTM! I'll let @ronuchit give the actual approval.

I agree that it's important to make clear that `DemonstrationResponses` are not the same as demonstrations for a task. Given the paradigm that @ashayathalye wants to use these in (i.e, interactively), I think it should be fine to not have a separate bit like `is_full_demo_for_task` or something, since I don't see why we might ever mix full demos with `DemonstrationResponses`.

File: src/structs.py
Comment (ronuchit) [inline src/structs.py:1129]: should we initialize the teacher with the training tasks instead, so that you only have to give `train_task_idx` here? i guess that's less explicit so maybe not?

File: src/teacher.py
Comment (ronuchit) [inline src/teacher.py:59]: to answer your comment on the PR, i think this is fine. anyone bold enough to use DemonstrationQuery/Response must be willing to accept the semantic consequences.

**Review by ronuchit — APPROVED** [review]
approving to unblock, but let's decide on my question before merging

File: src/structs.py
Comment (ronuchit) [reply src/structs.py:1129]: OTOH, it's slightly weird to give both of these fields...

Comment (tomsilver) [reply src/structs.py:1129]: yeah I think the weirdness of having both fields outweighs the explicitness concern. I'll try it that way

**Comment by tomsilver** [comment]
@ronuchit @NishanthJKumar ready for another look

File: src/teacher.py
Comment (ronuchit) [inline src/teacher.py:20]: it's funny how we've decided what the teacher gets in the init versus has to create for itself via `create_env` and the ensuing lines...

Comment (ronuchit) [reply src/teacher.py:20]: (not sure if any action item required, but it is weird)

File: tests/test_teacher.py
Comment (ronuchit) [inline tests/test_teacher.py:83]: does it matter that this was block0 before but block1 now?

**Review by ronuchit — APPROVED** [review]

File: src/teacher.py
Comment (tomsilver) [reply src/teacher.py:20]: I think it's okay. It's not "cheating" for the teacher to know the train tasks, vs knowing env.simulate, since the agent has train tasks as well.

File: tests/test_teacher.py
Comment (tomsilver) [reply tests/test_teacher.py]: no, but i changed it back to reduce entropy in the cosmos

**Pull request merged**
